pymatgen/symmetry/groups.py

```diff
@@ -223,13 +236,13 @@ def __init__(self, int_symbol: str) -> None:
         for spg in SpaceGroup.SYMM_OPS:
             if int_symbol in [spg["hermann_mauguin"], spg["universal_h_m"]]:
                 ops = [SymmOp.from_xyz_str(s) for s in spg["symops"]]
-                self.symbol = re.sub(r":", "", re.sub(r" ", "", spg["universal_h_m"]))
+                self.symbol = spg["hermann_mauguin"]
                 if int_symbol in SpaceGroup.sg_encoding:
                     self.full_symbol = SpaceGroup.sg_encoding[int_symbol]["full_symbol"]
                     self.point_group = SpaceGroup.sg_encoding[int_symbol]["point_group"]
                 else:
-                    self.full_symbol = re.sub(r" ", "", spg["universal_h_m"])
-                    self.point_group = spg["schoenflies"]
+                    self.full_symbol = spg["hermann_mauguin"]  # TODO full symbol not available from spg->maybe not add?
```

Comment:
can you elaborate on this TODO? does this mean the current `full_symbol` is incorrect or incomplete?

Reply:
Yes - as far as I can see, the full symbol (e.g., F4/m-32/m instead of Fm-3m) is only contained in SpaceGroup.sg_encoding, but not in SpaceGroup.SYMM_OPS. Both spg keys "hermann_mauguin" and "universal_h_m" do not contain this information. We could still set this attribute but issue a warning (maybe that's better than not setting it in these cases?)

Reply:
Added a warning and this seems to be only invoked when initializing non-standard settings (as expected). E.g., `SpaceGroup("Pmc2_1")` does not invoke a warning, but `SpaceGroup("P2_1ma")` does.

However, I struggle to write a test for that:
```python
    def test_full_symbol_warning(self):
        with pytest.warns(Warning,
                          match="Full symbol not available, falling back to short Hermann Mauguin symbol P2_1ma instead.") as warns:
            _ = SpaceGroup("P2_1ma")
        assert len(warns) == 1
```
This fails with `Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) were emitted. Emitted warnings: [].` and I don't understand why. Any ideas?

Reply:
are you sure you weren't testing against the PyPI release of `pymatgen`? your test passes for me. just pushed it to this PR. i expect it'll pass here as well

Reply:
Thanks for figuring out this issue! https://github.com/materialsproject/pymatgen/pull/3859/commits/a33cc3cf92e471865d132ca981d995bf87c8a8be